src/Nethermind/Nethermind.Merge.Plugin/Handlers/NewPayloadHandler.cs

```diff
@@ -118,7 +118,7 @@ public async Task<ResultWrapper<PayloadStatusV1>> HandleAsync(ExecutionPayload r
         if (!HeaderValidator.ValidateHash(block!.Header))
         {
             if (_logger.IsWarn) _logger.Warn(InvalidBlockHelper.GetMessage(block, "invalid block hash"));
-            return NewPayloadV1Result.Invalid(null, $"Invalid block hash {request.BlockHash}");
+            return NewPayloadV1Result.Invalid(null, $"Invalid block hash {request.BlockHash} does not match calculated hash {block.CalculateHash()}.");
```

Comment:
Maybe `ValidateHash` should have `out` parameter? There is one private with `ref`